project/ScalafixBuild.scala

```diff
@@ -15,7 +16,9 @@ import sbt.plugins.IvyPlugin
 
 object ScalafixBuild extends AutoPlugin with GhpagesKeys {
   override def trigger = allRequirements
-  override def requires = JvmPlugin && IvyPlugin
+  override def requires =
+    JvmPlugin &&
+      SbtVersionPolicyPlugin // don't let it override our mimaPreviousArtifacts
```

Comment:
By changing the order of plugin execution, this had a sneaky effect on publishing releases to sonatype as the `publishTo` below was effectively ignored 
```diff
 [info] publishTo
-[info] Some(FileRepository(sonatype-local-bundle, Patterns(ivyPatterns=Vector(), artifactPatterns=Vector(/home/brice/git/opensource/scalafix/target/sonatype-staging/0.10.2/[organisation]/[module](_[scalaVersion])(_[sbtVersion])/[revision]/[artifact]-[revision](-[classifier]).[ext]), isMavenCompatible=true, descriptorOptional=false, skipConsistencyCheck=false), FileConfiguration(true, None)))
+[info] Some(sonatype-staging: https://oss.sonatype.org/service/local/staging/deploy/maven2)
```

This prevents publishing releases
```
2022-09-20 22:12:38.036Z  info [SonatypeService] sonatypeRepository  : https://oss.sonatype.org/service/local  - (SonatypeService.scala:26)
2022-09-20 22:12:38.041Z  info [SonatypeService] sonatypeProfileName : ch.epfl.scala  - (SonatypeService.scala:27)
[info] Preparing a new staging repository for [sbt-sonatype] scalafix 0.10.3
2022-09-20 22:12:38.043Z  info [SonatypeClient] Reading staging repository profiles...  - (SonatypeClient.scala:108)
2022-09-20 22:12:40.251Z  info [SonatypeClient] Reading staging profiles...  - (SonatypeClient.scala:120)
2022-09-20 22:12:43.932Z  info [SonatypeClient] Creating a staging repository in profile ch.epfl.scala with a description key: [sbt-sonatype] scalafix 0.10.3  - (SonatypeClient.scala:126)
2022-09-20 22:12:51.705Z  info [SonatypeClient] Created successfully: chepflscala-2407  - (SonatypeClient.scala:139)
2022-09-20 22:13:00.349Z  info [SonatypeService] Found a previous staging repository [chepflscala-2405] status:open, profile:ch.epfl.scala(17f10a3293d844) description: [sbt-sonatype] scalafix 0.10.3  - (SonatypeService.scala:103)
2022-09-20 22:13:00.350Z  info [SonatypeClient] Dropping staging repository [chepflscala-2405] status:open, profile:ch.epfl.scala(17f10a3293d844) description: [sbt-sonatype] scalafix 0.10.3  - (SonatypeClient.scala:229)
2022-09-20 22:13:00.430Z  info [SonatypeService] Dropped successfully: chepflscala-2405  - (SonatypeService.scala:175)
2022-09-20 22:13:00.454Z error [Sonatype] 
java.io.IOException: Supplied file /home/runner/work/scalafix/scalafix/target/sonatype-staging/0.10.3 is a not an existing directory!
	at org.sonatype.spice.zapper.fs.AbstractDirectory.<init>(AbstractDirectory.java:32)
```
https://github.com/scalacenter/scalafix/actions/runs/3093699216/jobs/5006465269

https://github.com/scalacenter/scalafix/pull/1678#pullrequestreview-1114516786 should address that

Reply:
Do you think it’s a problem that comes from sbt-version-policy? Please let me know if I need to fix something there :)

Reply:
Sure, thanks for chiming in! The `publishTo` issue was unrelated, the build was clearly wrong, and the change in plugin order accidentally made it visible. We can follow up on https://github.com/scalacenter/sbt-version-policy/issues/138 though.